### PR TITLE
fix life-cycle script execution

### PR DESCRIPTION
**Pull request by alexlamsl**
- change current working directory for workspaces
- add `node_modules/.bin` to `PATH` before running

### CODE REVIEW

File: src/cli/run_command.zig [hunk 1]
```diff
@@ -533,82 +533,68 @@ pub const RunCommand = struct {
...
-                    defer needs_colon = true;
-                    if (needs_colon) {
-                        try new_path.append(':');
-                    }
-                    try new_path.appendSlice(bin_dirs[@intCast(usize, bin_dir_i)]);
+        {
+            var needs_colon = false;
+            if (package_json_dir.len > 0) {
+                defer needs_colon = true;
+                if (needs_colon) {
```
Comment (Jarred-Sumner) [inline src/cli/run_command.zig:568]: This branch will always be true

File: src/install/lockfile.zig [hunk 2]
```diff
@@ -112,7 +112,11 @@ const Stream = std.io.FixedBufferStream([]u8);
 pub const default_filename = "bun.lockb";
 
 pub const Scripts = struct {
-    const StringArrayList = std.ArrayListUnmanaged(string);
+    const Entry = struct {
```
Comment (Jarred-Sumner) [inline src/install/lockfile.zig:115]: we almost could add a --filter with this! Just missing another map to say "which package.json name does it belong to"

**Review by Jarred-Sumner — APPROVED** [review]

**Comment by Jarred-Sumner** [comment]
Looks good, thanks for fixing

File: src/cli/run_command.zig (hunk 1 above)
Comment (alexlamsl) [reply src/cli/run_command.zig:568]: would this `defer` be executed at the end of `if (package_json_dir.len > 0) {...}`? :thinking:

Comment (Jarred-Sumner) [reply src/cli/run_command.zig:568]: yeah

**Pull request merged**